climsight.py

```diff
@@ -370,14 +571,25 @@ def extract_climate_data(lat, lon, _hist, _future):
         soil = get_soil_from_api(lat, lon)
         st.markdown(f"**Soil type:** {soil}")
 
+        biodiversity = fetch_biodiversity(round(lat), round(lon))
+        biodiv_set = set()
+        for record in biodiversity['results']:
+            if record.get('taxonRank') != 'UNRANKED':
+                biodiv_set.add(record['scientificName'])
```

Comment:
`scientificName` gives us some strange looking references like `['Pristurus obsti Rösler & Wranik, 1999']` I would go for `biodiv_set.add(record['genericName']` instead.